[FG:InPlacePodVerticalScaling] Show CPU request in status only when configured

hshiina opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?

<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->
/kind bug

#### What this PR does / why we need it:
The minimum CPU request is set to a container internally if a CPU request is not configured. This minimum request does not need displaying to users. This PR sets a CPU request to a pod status only when a CPU request is configured in the pod spec.

This PR also prevents the internal minimum CPU request from triggering an unexpected resize in `computePodResizeAction()`.

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #126195

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
Fixed a bug that the internal minimum CPU request is displayed in a pod status when only a memory request is configured in the pod and the `InPlacePodVerticalScaling` feature is enabled.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

hshiina committed:
kubelet: Show CPU request in status only when configured

bart0sh commented on test/e2e_node/pod_resize_test.go:
```diff
@@ -1131,6 +1131,24 @@ func doPodResizeTests() {
+					Resources: &ContainerResources{MemReq: "300Mi"},
+				},
+			},
+		},
```
How about testing "CPU only" requests as well?

hshiina replied:
Thank you for the suggestion. I added a similar test case for CPU. In this case, CPU limit is necessary due to the issue #126388.

bart0sh commented:
/triage accepted
/priority important-longterm
/ok-to-test

hshiina committed:
e2e_node: Add test to resize container with cpu requests and limits

hshiina commented:
/test pull-kubernetes-node-kubelet-serial-podresize

hshiina commented:
/retest

hshiina commented:
It looks hitting a flake #126392.

hshiina committed:
kubelet: Do not trigger CPU resize by internal minimum requests

kubelet passes the minimum CPU requests (CPU shares) to the runtime if
CPU requests are not configured in a container in Linux. This minimum
value is contained in the container status got from the runtime. This
fix prevents this minimum request from triggering CPU resizing.

hshiina commented:
/test pull-kubernetes-node-kubelet-serial-podresize

hshiina commented:
/retest

hshiina commented:
/test pull-kubernetes-e2e-inplace-pod-resize-containerd-main-v2
/test pull-kubernetes-node-kubelet-serial-podresize

tallclair commented:
I think this PR is superseded by https://github.com/kubernetes/kubernetes/pull/128680

Please take a look at the linked PR.

/close
